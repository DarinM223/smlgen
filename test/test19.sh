../smlgen test19.sml no:s --recurmod --test
if cmp --silent test19.sml.test test19.sml.expected; then
  echo "test19 succeeded"
else
  echo "test19 failed"
  echo "Diff:"
  diff test19.sml.test test19.sml.expected
  exit 1
fi