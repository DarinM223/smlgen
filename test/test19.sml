structure Use =
struct
  datatype t =
    Use of {value: Value.t ref, user: Value.t, prev: t ref, next: t ref}
end

structure Value =
struct
  type id = int
  datatype value = Instr of id
  datatype t = Value of {id: id, value: value ref, uses: Use.t option ref}
end

structure Literal = struct end

structure Argument = struct end

structure Value = struct end
