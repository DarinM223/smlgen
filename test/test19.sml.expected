structure Component1 = struct datatype value = Instr of int end
structure Component2 =
struct
  datatype Use_t =
    Use of {value: Value_t ref, user: Value_t, prev: Use_t ref, next: Use_t ref}
  and Value_t =
    Value of {id: int, value: Component1.value ref, uses: Use_t option ref}
end
structure Use = struct type t = Component2.Use_t end

structure Value =
struct type id = int type value = Component1.value type t = Component2.Value_t end

structure Literal = struct end

structure Argument = struct end

structure Value = struct end